config-linux.md

```diff
@@ -189,6 +189,91 @@ In addition to any devices configured with this setting, the runtime MUST also s
 * [`/dev/ptmx`][pts.4].
   A [bind-mount or symlink of the container's `/dev/pts/ptmx`][devpts].
 
+## <a name="configLinuxNetworkDevices" />Network Devices
+
+Linux network devices are entities that send and receive data packets.
+They are not represented as files in the /dev directory, unlike block devices, network devices are represented with the [`net_device`][net_device] data structure in the Linux kernel.
+Network devices have their own network namespace and a set of operations distinct from regular file operations. Examples of network devices include Ethernet cards, loopback devices, and virtual devices like bridges, VLANs, and MACVLANs.
+
+This schema focuses solely on moving existing network devices identified by name into the container namespace. It does not cover the complexities of network device creation or network configuration, such as IP address assignment, routing, and DNS setup.
+
+**`netDevices`** (object, OPTIONAL) set of network devices that MUST be available in the container. The runtime is responsible for providing these devices; the underlying mechanism is implementation-defined.
+
+Only privileged containers with a dedicated network namespace can have network devices directly assigned to them. This is required because moving network devices requires CAP_NET_ADMIN capabilities, not present on rootless containers, and to ensure security and avoid conflicts manipulate interfaces in the runtime network namespace.
+The runtime MUST check that is possible to move the network interface to the container namespace and MUST [generate an error](runtime.md#errors) if the check fails.
+
+Notice that after deleting a network namespace, all its migratable network devices are moved to the default network namespace, unmoveable devices (NETIF_F_NETNS_LOCAL) and virtual devices (veth, macvlan, ...) are destroyed.
```

Comment:
Unmoveable devices can't be moved into a receiving netns in the first place, so no need to mention them here IMHO -- they're out of scope by definition.

I'm not sure to get why moveable virtual devices shouldn't be moved back into their original netns. IMO they should be treated like non-virtual devices.

Reply:
in many cases, virtual interfaces are destroyed automatically as soon as netns is destroyed. So, there is nothing to be moved back, or it might not be sensible to move back if e.g. macvlan/veth was pre-created specially for container.

Reply:
> in many cases, virtual interfaces are destroyed automatically as soon as netns is destroyed. So, there is nothing to be moved back

Right, but I assume netdevs should be moved back to their original netns _before_ their receiving netns is destroyed, so that should not be a problem.

> or it might not be sensible to move back if e.g. macvlan/veth was pre-created specially for container.

The runtime spec doesn't mention container restart -- it's just a delete + create. However, I could easily see use-cases where an higher-level container orchestrator would want to move that same interface back into a new container to implement a 'restart' operation.

Reply:
@akerouanton should virtual interface be required MUST be moved back or should I left it as MAY?